DasharoPayloadPkg/Library/PlatformBootManagerLib/PlatformBootManager.c

```diff
@@ -156,73 +170,21 @@ PlatformRegisterFvBootOption (
                   &NewOption, BootOptions, BootOptionCount
                   );
 
-  if (OptionIndex == -1) {
+  if (OptionIndex == -1 && FileExists) {

```

Comment:
I think this lacks a check for `Remove` to prevent adding an option on removal:
```suggestion
  if (OptionIndex == -1 && FileExists && !Remove) {
```

Reply:
Amended the commit: https://github.com/Dasharo/edk2/compare/f01ecabd9dc09421a7bc7309e229fbf5427d5218..a309360e147d1dd932950a06b12a0a594f21ea8b